### PR TITLE
implemented user rename feature

### PR DESCRIPTION
**Pull request by dracusds123**
Created the user rename feature for RWLDAPUserStores and JDBCUserstores:

Users will have a UID created when created or when first authenticated.

### CODE REVIEW

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java [hunk 1]
```diff
@@ -261,10 +335,10 @@ public boolean doPostAuthenticate(String userName, boolean authenticated,
                     Config emailConfig = null;
                     ConfigBuilder configBuilder = ConfigBuilder.getInstance();
                     try {
-                    	emailConfig =
-                    	              configBuilder.loadConfiguration(ConfigType.EMAIL,
-                    	                                              StorageType.REGISTRY,
-                    	                                              tenantId);
+                        emailConfig =
```
Comment (hpmtissera) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java]: format code to use the line size 120

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java [hunk 2]
```diff
@@ -486,7 +560,21 @@ public boolean doPreAddUser(String userName, Object credential, String[] roleLis
...
         // adding dto to thread local to be read again from the doPostAddUser method
         threadLocalProperties.get().put(USER_IDENTITY_DO, identityDTO);
+
+        isAddUserOperation.set(true);
+
         return true;
+
+    }
+
+    private String CreateUID(String userName) {
```
Comment (hpmtissera) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java]: use method naming convention

File: components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/UserProfileAdmin.java [hunk 3]
```diff
@@ -463,6 +474,64 @@ public UserProfileDTO getUserProfile(String username, String profileName)
...
+
+            PreparedStatement domainIdStatement = indbs.prepareStatement(domainIdQuery);
+
+            domainIdStatement.setString(1, domain);
+            domainIdStatement.setInt(2, tenantId);
+
+            ResultSet resultSet = domainIdStatement.executeQuery();
+
+            while (resultSet.next()) {
+                domainId = resultSet.getInt("UM_DOMAIN_ID");
```
Comment (hpmtissera) [inline components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/UserProfileAdmin.java]: define constants

Comment (hpmtissera) [inline components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/UserProfileAdmin.java]: Did you run sonar / Findbugs? We need to fix all the issues in new code.

Comment (dracusds123) [inline components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/UserProfileAdmin.java]: ran SONAR. Will fix the thing.. :D

On Mon, Jun 15, 2015 at 12:54 PM, Prasad Tissera <notifications@github.com>
wrote:

> In
> components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/UserProfileAdmin.java
> <https://github.com/wso2/carbon-identity/pull/435#discussion_r32396334>:
>
> > +
> > +        String domainIdQuery = "SELECT UM_DOMAIN_ID FROM UM_DOMAIN WHERE UM_DOMAIN_NAME=? AND UM_TENANT_ID=?";
> > +
> > +        try {
> > +
> > +            Connection indbs = JDBCPersistenceManager.getInstance().getDBConnection();
> > +
> > +            PreparedStatement domainIdStatement = indbs.prepareStatement(domainIdQuery);
> > +
> > +            domainIdStatement.setString(1, domain);
> > +            domainIdStatement.setInt(2, tenantId);
> > +
> > +            ResultSet resultSet = domainIdStatement.executeQuery();
> > +
> > +            while (resultSet.next()) {
> > +                domainId = resultSet.getInt("UM_DOMAIN_ID");
>
> Did you run sonar / Findbugs? We need to fix all the issues in new code.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wso2/carbon-identity/pull/435/files#r32396334>.
>



-- 
-Damith -

**Pull request closed without merging**